Code/Tools/ProjectManager/Source/GemCatalog/GemModel.cpp

```diff
@@ -479,6 +483,28 @@ namespace O3DE::ProjectManager
         return previouslyAdded && !added;
     }
 
+    void GemModel::DeactivateDependentGems(QAbstractItemModel& model, const QModelIndex& modelIndex)
+    {
+        GemModel* gemModel = GetSourceModel(&model);
+        AZ_Assert(gemModel, "Failed to obtain GemModel");
+
+        QVector<QModelIndex> dependentGems = gemModel->GatherDependentGems(modelIndex);
+        if (!dependentGems.isEmpty())
+        {
+            // we need to deactivate all gems that depend on this one
+            for (auto dependentModelIndex : dependentGems)
+            {
+                DeactivateDependentGems(model, dependentModelIndex);
+            }
+
+        }
+        else
+        {
+            // Deactivate this gem
+            SetIsAdded(model, modelIndex, false);
```

Comment:
Why are we deactivating this gem here and not just the dependents?  And if we should deactivate the gem here, shouldn't we do it no matter if it has dependent gems or not?

Reply:
Also, the GatherDependentGems should return a flattened deep dependency list, so you shouldn't need to recurse.